src/hotspot/share/cds/filemap.cpp

```diff
@@ -111,14 +111,14 @@ void FileMapInfo::fail_stop(const char *msg, ...) {
 void FileMapInfo::fail_continue(const char *msg, ...) {
   va_list ap;
   va_start(ap, msg);
-  fail_continue_impl(LogLevel::Info, msg, ap);
+  fail_continue_impl(LogLevel::Warning, msg, ap);
   va_end(ap);
 }
 
-void FileMapInfo::fail_continue(LogLevelType level, const char *msg, ...) {
+void FileMapInfo::fail_continue_nowarn(const char *msg, ...) {
```

Comment:
It is better to name a method by what it does do rather than what it doesn't. 

But I am finding the naming here problematic anyway as we do not always continue - the logic is basically `possibly_fail_else_log_and_continue`. Maybe the simplest thing here is to only have the variant that takes the `LogLevel` and so every callsite is explicit about whether it logs at warning or info level rather than try to accommodate warn/info into the name somehow?

Reply:
I agree that the `fail_continue` is not the ideal name for what this is trying to accomplish: log a message and *possibly* fail. 

That being said, I felt it was appropriate to abstract away the need to manage `LogLevel` as only Info and Warning are used with `fail_continue`. If someone does want to use any of the other channels, like debug, error, etc, they can still use the `fail_continue` variant that excepts a `LogLevel`, but they wouldn't have to worry about it otherwise. I think the binary option of warning vs no warning can help to keep it simple.

Reply:
In that case I return to "It is better to name a method by what it does do rather than what it doesn't." So `fail_continue` and `fail_continue_warn`?

Reply:
I think the methods can be renamed to

```
FileMapInfo::notify_failure(LogLevelType level, const char *msg, ...);
FileMapInfo::notify_failure(const char *msg, ...); // use warning channel
```

In most cases we use the warning channel, so that should be the default.

The naming of `notify_failure()` follows existing functions such as `VM_GC_Operation::notify_gc_begin()` to indicate that something has happened and some action will happen as a result.

In our case, it's reasonable to expect that some logging may happen as a result of the failure, and the VM may even exit.

By the way, the VM exits only if `-Xshare:on` is specified in the command-line. As specified in https://docs.oracle.com/en/java/javase/17/vm/class-data-sharing.html,  `-Xshare:on` is indented for testing only and should not be used in production.

Reply:
To be in line with @dholmes-ora's suggestion, would we use the names `notify_failure` and `notify_failure_warn` ? That seems a bit redundant and ambiguous to me, so I propose `notify_failure` for the warning channel and `log_failure` for the info channel and other possible levels.

Reply:
Actually, after thinking about this more, I think we should rewrite all the calls to `FileMapInfo::fail_continue()` with `log_xxx()` calls. This will bring the CDS logging code up-to-date and we can get rid of the convoluted `fail_continue()` function.

`fail_continue()` was ancient code  from the days when Unified Logging wasn't available. It currently does three things:

- (a) If `PrintSharedArchiveAndExit` or `RequireSharedSpaces` are enabled, always log the message so it's visible to the user
- (b) If  `RequireSharedSpaces` is enabled, exit the VM immediately
- Otherwise, log the message to the `cds` channel.

As mentioned in [JDK-8292269](https://bugs.openjdk.org/browse/JDK-8292269), we want to log some of the cds failure messages to the `info` channel, to avoid warning messages when the failure is expected (i.e., when `-XX:-UseCompressedClassPointers` is specified). My proposal is change all the `fail_continue()` calls to `log_info(cds)` or `log_warning(cds)`

The reason for (a) is:  `PrintSharedArchiveAndExit` is for diagnosing problems with the CDS archive; `RequireSharedSpaces`, which is driven by the `-Xshare:on` command-line flag, which should be used only in the regression test suite (see [here](https://docs.oracle.com/en/java/javase/17/vm/class-data-sharing.html)). In both these cases, we can simply force the `cds` logging level to `info` inside arguments.cpp (using `LogConfiguration::configure_stdout()`).

For (b), we don't have to exit immediately when the error is detected. We can exit the VM in `MetaspaceShared::initialize_runtime_shared_and_meta_spaces()` when `result == MAP_ARCHIVE_SUCCESS`.